python/ray/data/_internal/progress_bar.py

```diff
@@ -87,7 +87,7 @@ def set_description(self, name: str) -> None:
             self._bar.set_description(name)
 
     def update(self, i: int) -> None:
-        if self._bar:
+        if self._bar and i != 0:
```

Comment:
Why is this change needed?

Reply:
This avoids progress bar update spam where the bar isn't actually moving (it cleans up non interactive shells).